tests/datasources/test_chirps.py

```diff
@@ -0,0 +1,487 @@
+import pytest
+import xarray as xr
+from xarray.coding.cftimeindex import CFTimeIndex
+
+from aatoolbox import ChirpsDaily, ChirpsMonthly, GeoBoundingBox
+
+DATASOURCE_BASE_DIR = "chirps"
+_START_YEAR = 2020
+_START_MONTH = 7
+_START_DAY = 15
+_END_YEAR = 2020
+_END_MONTH = 8
+_END_DAY = 1
+_END_YEAR_FUTURE = 2100
+_END_MONTH_FUTURE = 10
+_END_DAY_FUTURE = 2
+
+_START_MONTH_STR = calendar.month_abbr[_START_MONTH]
+_END_MONTH_STR = calendar.month_abbr[_END_MONTH]
+_END_MONTH_FUTURE_STR = calendar.month_abbr[_END_MONTH_FUTURE]
+
+
+@pytest.fixture
+def mock_chirps(mock_country_config):
+    """Create Chirps class with mock country config."""
+    geo_bounding_box = GeoBoundingBox(
+        lat_max=6, lat_min=3.2, lon_max=2, lon_min=-3
+    )
+
+    def _mock_chirps(frequency: str = "daily", resolution: float = 0.05):
+        if frequency == "daily":
+            chirps = ChirpsDaily(
+                country_config=mock_country_config,
+                geo_bounding_box=geo_bounding_box,
+            )
+        else:
+            chirps = ChirpsMonthly(
+                country_config=mock_country_config,
+                geo_bounding_box=geo_bounding_box,
+            )
+        return chirps
+
+    return _mock_chirps
+
+
+def test_valid_arguments_class(mock_country_config):
+    """Test for resolution in initialisation class."""
+    geo_bounding_box = GeoBoundingBox(
+        lat_max=6, lat_min=3.2, lon_max=2, lon_min=-3
+    )
+    with pytest.raises(ValueError):
+        ChirpsDaily(
+            country_config=mock_country_config,
+            geo_bounding_box=geo_bounding_box,
+            resolution=0.10,
+        )
+
+
+def test_switching_resolution_class(mock_country_config):
+    """Test for switching instance resolution for monthly class."""
+    geo_bounding_box = GeoBoundingBox(
+        lat_max=6, lat_min=3.2, lon_max=2, lon_min=-3
+    )
+    chirps = ChirpsMonthly(
+        country_config=mock_country_config,
+        geo_bounding_box=geo_bounding_box,
+        resolution=0.25,
+    )
+    assert chirps._resolution == 0.05
+
+
+@pytest.fixture
+def mock_download(mocker, mock_chirps):
+    """Create mock for download method."""
+    download_mock = mocker.patch(
+        "aatoolbox.datasources.chirps.chirps._Chirps._actual_download"
+    )
+
+    def _mock_download(
+        frequency: str,
+        start_year,
+        end_year,
+        start_month,
+        end_month,
+        start_day,
+        end_day,
+    ):
+        chirps = mock_chirps(frequency=frequency)
+        chirps.download(
+            start_year=start_year,
+            end_year=end_year,
+            start_month=start_month,
+            end_month=end_month,
+            start_day=start_day,
+            end_day=end_day,
+        )
+        _, kwargs_download = download_mock.call_args
+        url = kwargs_download["url"]
+        filepath = kwargs_download["filepath"]
+        return url, filepath
+
+    return _mock_download
+
+
+def test_download_monthly(
```

Comment:
In this test and the next few, I think you still need to add a mock out `_get_last_available_date`